src/Microsoft.Extensions.Testing.Abstractions/PortablePdbReader.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+
+namespace Microsoft.Extensions.Testing.Abstractions
+{
+    public class PortablePdbReader : IPdbReader
+    {
+        private MetadataReader _reader;
+
+        public PortablePdbReader(Stream pdbStream)
+        {
+            pdbStream.Position = 0;
+            var buffer = new byte[pdbStream.Length];
+            pdbStream.Read(buffer, 0, buffer.Length);
+
+            unsafe
+            {
+                fixed (byte* bufferPtr = buffer)
+                {
+                    _reader = new MetadataReader(bufferPtr, buffer.Length);
```

Comment:
@tmat Are you referring to something like this:

_gcHandle = GCHandle.Alloc(buffer);
            unsafe
            {
                _reader = new MetadataReader((byte*)GCHandle.ToIntPtr(_gcHandle).ToPointer(), buffer.Length);
            }

followed by:

public void Dispose() { _gcHandle.Free() }.

Also, I couldn't figure out how to get rid of the unsafe code. All MetadataReader constructors take a byte*.